pkg/component/autoscaling/vpa/admissioncontroller.go

```diff
@@ -140,21 +145,40 @@ func (v *vpa) reconcileAdmissionControllerService(service *corev1.Service) {
 	switch v.values.ClusterType {
 	case component.ClusterTypeSeed:
 		metav1.SetMetaDataAnnotation(&service.ObjectMeta, resourcesv1alpha1.NetworkingFromWorldToPorts, fmt.Sprintf(`[{"protocol":"TCP","port":%d}]`, vpaconstants.AdmissionControllerPort))
+		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForSeedScrapeTargets(service, networkingv1.NetworkPolicyPort{
+			Port:     ptr.To(intstr.FromInt32(admissionControllerMetricsPort)),
+			Protocol: ptr.To(corev1.ProtocolTCP),
+		}))
 	case component.ClusterTypeShoot:
 		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForWebhookTargets(service, networkingv1.NetworkPolicyPort{
 			Port:     ptr.To(intstr.FromInt32(vpaconstants.AdmissionControllerPort)),
 			Protocol: ptr.To(corev1.ProtocolTCP),
 		}))
+		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForScrapeTargets(service, networkingv1.NetworkPolicyPort{
+			Port:     ptr.To(intstr.FromInt32(admissionControllerMetricsPort)),
+			Protocol: ptr.To(corev1.ProtocolTCP),
+		}))
+		metav1.SetMetaDataAnnotation(&service.ObjectMeta, resourcesv1alpha1.NetworkingPodLabelSelectorNamespaceAlias, v1beta1constants.LabelNetworkPolicyShootNamespaceAlias)
```

Comment:
IIUC this annotation is needed for the vpa-recommender due to https://github.com/gardener/gardener/blob/cd0992b79d2c8bfa9d2c12a2e4a0dba47c10bc3f/pkg/component/autoscaling/vpa/recommender.go#L296-L298

But why it is needed for the vpa-admission-controller? I guess the vpa-admission-controller is only scraped by the prometheus in the corresponding namespace, right? If it is not needed, we can drop it to prevent some NetworkPolicy creations.

Reply:
That's correct, done.